lua/telescope/make_entry.lua

```diff
@@ -612,31 +612,41 @@ function make_entry.gen_from_packages(opts)
   end
 end
 
-function make_entry.gen_from_apropos()
+function make_entry.gen_from_apropos(opts)
+  local sections = {}
+  for _, section in ipairs(vim.split(opts.sections, ',', true)) do
+    sections[section] = true
+  end
+
   local displayer = entry_display.create {
-    separator = "",
+    separator = ' ',
     items = {
-      { width = 30 },
-      { remaining = true },
+      {},
+      {},
     },
   }
 
   local make_display = function(entry)
+    -- Stopgap measure to set `width` for this field. See #414
```

Comment:
We should actually fix it rather than doing that hack.
One thing we can do is pass in like `truncate = false` which will disable `truncate`.

Reply:
I found way to solve #414 fundamentally. I rewrote `strdisplaywidth()` with `ffi` module and enable to use `width` feature in vim.loop.

https://github.com/delphinus/telescope.nvim/commit/f5d6c98293dc25a59d88811c55e29a9934a340ec

I will test more and apple this to this PR.

Reply:
First of all i love the idea :) Good job

Now three other things:
- I am concerned about windows. I don't know `linetabsize_col`, so is it a vim function we can always call on all platforms or is it a unix function? 
- Do we wanna support not jit at all. Every time i do something with `ffi` i currently just ignore the fact. Telescope without jit is not fast enough anyway. Just thinking out load
- i think you don't need to create the `c_` types you should just be able to pass the lua int and lua string. I think that just works for primitive types. At least thats what i am doing 😆 And thats what the documentation does: https://luajit.org/ext_ffi.html

Reply:
> * I am concerned about windows. I don't know linetabsize_col, so is it a vim function we can always call on all platforms or is it a unix function?

I just converted the logic of `strdisplaywidth()` from Neovim's source. `linetabsize_col` is in it. It should work everywhere Neovim works. ;)

https://github.com/neovim/neovim/blob/a1ed941a7881122fda2fd48e71e890ed55e4d08e/src/nvim/eval/funcs.c#L9845-L9858

For other 2 points, I will reply later. 👍

Reply:
I thought so. Thanks for clarifying :)

Reply:
> * Do we wanna support not jit at all. Every time i do something with ffi i currently just ignore the fact. Telescope without jit is not fast enough anyway. Just thinking out load. Nvm we also doing if jit in path.shorten. Didn't know that

I see. I'm not familiar with LuaJIT affairs in Neovim, but I think the fallback measure `#str - col` is enough for ordinary latin1 characters. I'll use that.

> * i think you don't need to create the c_ types you should just be able to pass the lua int and lua string. I think that just works for primitive types. At least thats what i am doing 😆 And thats what the documentation does: https://luajit.org/ext_ffi.html

I found `c_col` is unnecessary as you say, but still necessary for `c_str`.

```lua
local ffi = require('ffi')

ffi.cdef[[
  typedef unsigned char char_u;
  int linetabsize_col(int startcol, char_u *s);
]]

local col = 0
local str = 'foobar'
print( ffi.C.linetabsize_col(col, str) - col )
--local c_str = ffi.new('char[?]', #str + 1)
--ffi.copy(c_str, str)
--print( ffi.C.linetabsize_col(col, c_str) - col )
```

```vim
E5113: Error while calling lua chunk: scratch/strdisplaywidth.lua:10: bad argument #2 to 'linetabsize_col' (cannot convert 'string' to 'unsigned char *')
```

----

I'm thinking for posting another PR for this. I found other problems for `strdisplaywidth` 🤔

Reply:
Created #413. I'll try to solve problems for multi-byte strings later in another PR...

Reply:
> but still necessary for c_str

Oh yeah it works for `const char*` but not for `unsigned char*`

Reply:
This will be fixed in #464.

Reply:
Fixed on 7de61bc